src/audio/module_adapter/module_adapter.c

```diff
@@ -524,12 +531,32 @@ int module_adapter_copy(struct comp_dev *dev)
 	struct comp_buffer *sink;
 	struct processing_module *mod = comp_get_drvdata(dev);
 	struct module_data *md = &mod->priv;
+	struct comp_copy_limits c;
 	struct list_item *blist;
 	size_t size = MAX(mod->deep_buff_bytes, mod->period_bytes);
 	uint32_t min_free_frames = UINT_MAX;
 	int ret, i = 0;
 
-	/* get the least number of free frames from all sinks */
+	source = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
+	sink = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
+
+	comp_info(dev, "module_adapter_copy(): start");
+
+	/*
+	 * Simplify calculation of bytes_to_process for modules that produce period_bytes every
+	 * period and have only 1 source and 1 sink buffer
+	 */
+	if (mod->simple_copy) {
+		comp_get_copy_limits_with_lock_frame_aligned(source, sink, &c);
+		buffer_stream_invalidate(source, c.frames * c.source_frame_bytes);
+
+		/* note that the size is in number of frames not the number of bytes */
+		mod->input_buffers[0].size = c.frames;
+		mod->input_buffers[0].consumed = 0;
+		mod->output_buffers[0].size = 0;
+		goto process;
```

Comment:
I'd rather do an `else` even if it increases the patch size with no actual changes

Reply:
> @lyakh I thought you were a kernel guy ? I'm fine with goto :)

@lgirdwood exactly, I got used to the kernel programming style and got to like it. And here's a quote from Documentation/process/coding-style.rst:
```
The goto statement comes in handy when a function exits from multiple
locations and some common work such as cleanup has to be done.  If there is no
cleanup needed then just return directly.
```
So, I interpret that as it's recommended to only use `goto` for error handling / clean up. I know that this recommendation isn't followed strictly everywhere in the kernel and in SOF, so, the world won't end if we commit yet one more of them, but personally I'd find it prettier with an `else` here. My personal preference only.